### PR TITLE
Add Git guide for team members

### PR DESCRIPTION
**Pull request by alihussain03**
Add Git guide first version

### CODE REVIEW

File: GitGuide.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   5. [Add SSH keys in GITHUB](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh)
+2. It good to have some git desktop client to work fast. it is optional and totally depends upon the individual. Some recommendation are list down
+   1. [Github Desktop](https://desktop.github.com/) 
+   2. [Source Tree](https://www.sourcetreeapp.com/)
+   3. [Tortoise Git](https://tortoisegit.org/)
+## GitHub Flow Overview 
+   1. clone repository from GitHub 
+   2. create a working branch on local computer
+   3. Implement the requested feature, make sure all tests are passing, and commit all changes in the new branch
+   4. checkout the master branch locally
```
Comment (rvsape) [inline GitGuide.md]: master -> main

File: GitGuide.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+   1. [Github Desktop](https://desktop.github.com/) 
+   2. [Source Tree](https://www.sourcetreeapp.com/)
+   3. [Tortoise Git](https://tortoisegit.org/)
+## GitHub Flow Overview 
+   1. clone repository from GitHub 
+   2. create a working branch on local computer
+   3. Implement the requested feature, make sure all tests are passing, and commit all changes in the new branch
+   4. checkout the master branch locally 
+   5. Merge master branch into your local branch    
+   6. Remove merge conflicts if occurs
```
Comment (rvsape) [inline GitGuide.md]: remove -> resolve

File: GitGuide.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+   7. push/upload the changes to remote branch
+   8. Create Pull request
+   9. Add other as reviewers
+   10. Merge branch into master if everyone approve the changes
```
Comment (rvsape) [inline GitGuide.md]: master -> main

File: GitGuide.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    1. Don’t git push straight to master. Make your own branch and work 
+    2. Don’t commit code as an unrecognized author. Configure your email, username properly
+    3. Add other team memebers for faster code reviews
+    4. Don’t add secrets into source code
+    5. Don’t commit dependencies into source control
+    6. Don’t commit local config files into source control 
+    7. Archive dead repositories/branches. if no repo or bracnh is more beneficial, archive it properly
+    8. Use a branch naming convention
+    9. Keep branches up to date with master
+    10. keep pushing your code to master on daily base
```
Comment (rvsape) [inline GitGuide.md]: this conflicts with don't push straight to main

File: GitGuide.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+   6. Remove merge conflicts if occurs
+   7. push/upload the changes to remote branch
+   8. Create Pull request
+   9. Add other as reviewers
+   10. Merge branch into master if everyone approve the changes 
+   11. ![Github Flow](GitHubFlow.PNG)
+
+
+## Best practices
+    1. Don’t git push straight to master. Make your own branch and work
```
Comment (rvsape) [inline GitGuide.md]: master -> main

File: GitGuide.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+
+## Best practices
+    1. Don’t git push straight to master. Make your own branch and work 
+    2. Don’t commit code as an unrecognized author. Configure your email, username properly
+    3. Add other team memebers for faster code reviews
+    4. Don’t add secrets into source code
+    5. Don’t commit dependencies into source control
+    6. Don’t commit local config files into source control 
+    7. Archive dead repositories/branches. if no repo or bracnh is more beneficial, archive it properly
```
Comment (rvsape) [inline GitGuide.md:35]: could possibly just delete branch after merge

**Comment by OpaJaa** [comment]
Seems simple enough guide. Don't know how familiar everyone is with git, so maybe there could be example cli commands for different tasks?

And could we have a common .gitignore to avoid some of the points in "Best practices"?

**Review by OpaJaa — COMMENTED** [review]
Reviewed. Comments in conversation.

File: GitGuide.md (hunk 1 above)
Comment (alihussain03) [reply GitGuide.md]: Done

File: GitGuide.md (hunk 2 above)
Comment (alihussain03) [reply GitGuide.md]: Done

File: GitGuide.md (hunk 3 above)
Comment (alihussain03) [reply GitGuide.md]: Done

File: GitGuide.md (hunk 5 above)
Comment (alihussain03) [reply GitGuide.md]: Done

File: GitGuide.md (hunk 6 above)
Comment (alihussain03) [reply GitGuide.md:35]: Done

**Comment by alihussain03** [comment]
@OpaJaa i have added one link for common commands. it will be difficult to add all commands here and maybe it will be rework. Regarding gitignore file, as it will be js project so we can use this file and update accordingly https://gist.github.com/andreasonny83/b24e38b7772a3ea362d8e8d238d5a7bc

**Comment by OpaJaa** [comment]
> 

A link is ok. I was thinking like git checkout main, git branch cc-feature-1 / git checkout cc-feature-1 etc. Don't know if people even use cli or if git is well known for everyone. 

Also that .gitignore looks promising.

**Review by OpaJaa — DISMISSED** [review]

**Review by rvsape — DISMISSED** [review]

File: GitGuide.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+   2. [Source Tree](https://www.sourcetreeapp.com/)
+   3. [Tortoise Git](https://tortoisegit.org/)
+## GitHub Flow Overview 
+   1. clone repository from GitHub 
+   2. create a working branch on local computer
+   3. Implement the requested feature, make sure all tests are passing, and commit all changes in the new branch
+   4. checkout the main branch locally 
+   5. Merge main branch into your local branch    
+   6. Resolve merge conflicts if occurs
+   7. push/upload the changes to remote branch
```
Comment (bunjovis) [inline GitGuide.md]: Isn't this flow now pushing straight to main branch? Should it be something like, clone repo, create new branch, implement stuff, commit and push to remote, create pull request, resolve merge conflicts. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github here is a good link how you can use github to resolve conflicts.

File: GitGuide.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+   2. [Set your username in GIT](https://docs.github.com/en/get-started/getting-started-with-git/setting-your-username-in-git)
+   3. [Set your commit email address in Git](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address)
+   4. [Authenticating with GitHub from Git](https://docs.github.com/en/get-started/quickstart/set-up-git#authenticating-with-github-from-git)
+   5. [Add SSH keys in GITHUB](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh)
+2. It good to have some git desktop client to work fast. it is optional and totally depends upon the individual. Some recommendation are list down
+   1. [Github Desktop](https://desktop.github.com/) 
+   2. [Source Tree](https://www.sourcetreeapp.com/)
+   3. [Tortoise Git](https://tortoisegit.org/)
+## GitHub Flow Overview 
+   1. clone repository from GitHub
```
Comment (bunjovis) [inline GitGuide.md:14]: Clone repository / pull latest to main

File: GitGuide.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+   1. clone repository from GitHub 
+   2. create a working branch on local computer
+   3. Implement the requested feature, make sure all tests are passing, and commit all changes in the new branch
+   4. checkout the main branch locally 
+   5. Merge main branch into your local branch    
+   6. Resolve merge conflicts if occurs
+   7. push/upload the changes to remote branch
+   8. Create Pull request
+   9. Add other as reviewers
+   10. Merge branch into main if everyone approve the changes
```
Comment (bunjovis) [inline GitGuide.md]: I would suggest dropping "if everyone" to "if one person". What do others think?

File: GitGuide.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+   3. [Tortoise Git](https://tortoisegit.org/)
+## GitHub Flow Overview 
+   1. clone repository from GitHub 
+   2. create a working branch on local computer
+   3. Implement the requested feature, make sure all tests are passing, and commit all changes in the new branch
+   4. checkout the main branch locally 
+   5. Merge main branch into your local branch    
+   6. Resolve merge conflicts if occurs
+   7. push/upload the changes to remote branch
+   8. Create Pull request
```
Comment (bunjovis) [inline GitGuide.md]: Create pull request to main branch

File: GitGuide.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+   9. Add other as reviewers
+   10. Merge branch into main if everyone approve the changes 
+   11. ![Github Flow](GitHubFlow.PNG)
+
+
+## Best practices
+    1. Don’t git push straight to main. Make your own branch and work 
+    2. Don’t commit code as an unrecognized author. Configure your email, username properly
+    3. Add other team memebers for faster code reviews
```
Comment (bunjovis) [inline GitGuide.md]: members

File: GitGuide.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+## GitHub Flow Overview 
+   1. clone repository from GitHub 
+   2. create a working branch on local computer
+   3. Implement the requested feature, make sure all tests are passing, and commit all changes in the new branch
+   4. checkout the main branch locally 
+   5. Merge main branch into your local branch    
+   6. Resolve merge conflicts if occurs
+   7. push/upload the changes to remote branch
+   8. Create Pull request
+   9. Add other as reviewers
```
Comment (bunjovis) [inline GitGuide.md]: others

File: GitGuide.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+## Best practices
+    1. Don’t git push straight to main. Make your own branch and work 
+    2. Don’t commit code as an unrecognized author. Configure your email, username properly
+    3. Add other team memebers for faster code reviews
+    4. Don’t add secrets into source code
+    5. Don’t commit dependencies into source control
+    6. Don’t commit local config files into source control 
+    7. Archive dead repositories/branches. if no repo or bracnh is more beneficial, archive it properly
+    8. Use a branch naming convention
```
Comment (bunjovis) [inline GitGuide.md]: Which convention? I suggest naming after features, for example dev-somecomponent, dev-bugfixes, dev-document.

File: GitGuide.md (hunk 7 above)
Comment (alihussain03) [reply GitGuide.md]: no it is not pushing to main branch. i have updated the line 7.

File: GitGuide.md (hunk 9 above)
Comment (alihussain03) [reply GitGuide.md]: i think it should be everyone. you can add limited persons as reviewers but every reviewer should approve that

File: GitGuide.md (hunk 10 above)
Comment (alihussain03) [reply GitGuide.md]: done

File: GitGuide.md (hunk 11 above)
Comment (alihussain03) [reply GitGuide.md]: done

File: GitGuide.md (hunk 12 above)
Comment (alihussain03) [reply GitGuide.md]: done

File: GitGuide.md (hunk 13 above)
Comment (alihussain03) [reply GitGuide.md]: this is what we can decide in meeting. btw it is article to read https://codingsight.com/git-branching-naming-convention-best-practices/

File: GitGuide.md (hunk 9 above)
Comment (OpaJaa) [reply GitGuide.md]: I'd say not everyone, but maybe there could be some division on who reviews whose code. I mean like backend devs review code of other backend devs or something. And maybe the lead developer should review all code?  But after that everyone of those should accept the changes. There might be a lot of code to review if everyone should review and accept everything all the time.

Comment (bunjovis) [reply GitGuide.md]: Lets discuss this on thursdays meeting

File: GitGuide.md (hunk 13 above)
Comment (bunjovis) [reply GitGuide.md]: lets discuss this on thursdays meeting

File: GitGuide.md [hunk 14]
```diff
@@ -0,0 +1,45 @@
+# Git Guide 
+## Prerequisite
+1. At the heart of GitHub is an open source version control system (VCS) called Git. Git is responsible for everything GitHub-related that happens locally on your computer. 
+   1. [Download and install the latest version of GIT](https://git-scm.com/downloads)
+   2. [Set your username in GIT](https://docs.github.com/en/get-started/getting-started-with-git/setting-your-username-in-git)
+   3. [Set your commit email address in Git](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address)
+   4. [Authenticating with GitHub from Git](https://docs.github.com/en/get-started/quickstart/set-up-git#authenticating-with-github-from-git)
+   5. [Add SSH keys in GITHUB](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh)
+2. It good to have some git desktop client to work fast. it is optional and totally depends upon the individual. Some recommendation are list down
```
Comment (hannakortetjarvi) [inline GitGuide.md:9]: Some corrections to grammar:
1) It is* good to have
2) Some recommendations* are listed* down

File: GitGuide.md [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+
+## Best practices
+
+    1. Don’t git push straight to main. Make your own branch and work 
+    2. Don’t commit code as an unrecognized author. Configure your email, username properly
+    3. Add other team members for faster code reviews
+    4. Don’t add secrets into source code
+    5. Don’t commit dependencies into source control
+    6. Don’t commit local config files into source control 
+    7. Archive dead repositories/branches. if no repo or bracnh is more beneficial, archive it properly
```
Comment (hannakortetjarvi) [inline GitGuide.md:35]: branch*

**Review by olivena — DISMISSED** [review]

**Review by rvsape — DISMISSED** [review]

**Review by bunjovis — APPROVED** [review]

**Review by OpaJaa — APPROVED** [review]

**Pull request merged**